src/OpenTelemetry.Extensions.PersistentStorage/PersistentStorageEventSource.cs

```diff
@@ -16,92 +16,180 @@
 
 using System;
 using System.Diagnostics.Tracing;
-using System.Runtime.CompilerServices;
+using System.Globalization;
+using System.Threading;
 
 namespace OpenTelemetry.Extensions.PersistentStorage
 {
     [EventSource(Name = EventSourceName)]
     internal sealed class PersistentStorageEventSource : EventSource
     {
         public static PersistentStorageEventSource Log = new PersistentStorageEventSource();
-        private const string EventSourceName = "OpenTelemetry-Contrib-PersistentStorage";
+        private const string EventSourceName = "OpenTelemetry-Extensions-PersistentStorage";
 
         [NonEvent]
-        public void Critical(string message, object value = null)
+        public void CouldNotReadFileBlob(string filePath, Exception ex)
         {
-            this.Write(EventLevel.Critical, message, value);
+            if (this.IsEnabled(EventLevel.Informational, EventKeywords.All))
+            {
+                this.CouldNotReadFileBlob(filePath, ToInvariantString(ex));
+            }
+        }
+
+        [NonEvent]
+        public void CouldNotWriteFileBlob(string filePath, Exception ex)
+        {
+            if (this.IsEnabled(EventLevel.Informational, EventKeywords.All))
+            {
+                this.CouldNotWriteFileBlob(filePath, ToInvariantString(ex));
+            }
         }
 
         [NonEvent]
-        public void Error(string message, object value = null)
+        public void CouldNotLeaseFileBlob(string filePath, Exception ex)
         {
-            this.Write(EventLevel.Error, message, value);
+            if (this.IsEnabled(EventLevel.Informational, EventKeywords.All))
+            {
+                this.CouldNotLeaseFileBlob(filePath, ToInvariantString(ex));
+            }
         }
 
         [NonEvent]
-        public void Warning(string message, object value = null)
+        public void CouldNotDeleteFileBlob(string filePath, Exception ex)
         {
-            this.Write(EventLevel.Warning, message, value);
+            if (this.IsEnabled(EventLevel.Informational, EventKeywords.All))
+            {
+                this.CouldNotDeleteFileBlob(filePath, ToInvariantString(ex));
+            }
         }
 
         [NonEvent]
-        public void Informational(string message, object value = null)
+        public void CouldNotCreateFileBlob(Exception ex)
         {
-            this.Write(EventLevel.Informational, message, value);
+            if (this.IsEnabled(EventLevel.Informational, EventKeywords.All))
+            {
+                this.CouldNotCreateFileBlob(ToInvariantString(ex));
+            }
+        }
+
+        [NonEvent]
+        public void CouldNotRemoveExpiredBlob(string filePath, Exception ex)
+        {
+            if (this.IsEnabled(EventLevel.Informational, EventKeywords.All))
```

Comment:
This looks wrong.
Line 78 is checking if `Informational` is enabled.
But the actual event method is `Warning` line 141.

Looks like some of the other `NonEvent` methods have the same mistake.

Reply:
My bad: I will fix it.

Reply:
fixed